File: 0220/1221.py
import sys
sys.stdin = open('1221.txt', 'r')
sys.stdout = open('1221_out.txt', 'w')
T = int(input())
for _ in range(T):
    test_case, n = list(input().split())
    n = int(n)
    result = {
        'ZRO': 0,
        'ONE': 0,
        'TWO': 0,
        'THR': 0,
        'FOR': 0,
        'FIV': 0,
        'SIX': 0,
        'SVN': 0,
        'EGT': 0,
        'NIN': 0
    }
    re = ['ZRO', 'ONE', 'TWO', 'THR', 'FOR', 'FIV', 'SIX', 'SVN', 'EGT', 'NIN']
    li = list(input().split())
    for i in range(n):
        result[li[i]] += 1
    i = j = 0
    print(test_case)
    while i < n:
        if result[re[j]] != 0:
            result[re[j]] -= 1
            print(re[j], end=' ')
            i += 1
        else:
            j += 1
    print()

File: swea/c++/1221.cpp
#include<iostream>
#include<algorithm>
using namespace std;

char number[10][4] = { "ZRO", "ONE", "TWO", "THR", "FOR", "FIV", "SIX", "SVN", "EGT", "NIN" };
int num_cnt[10] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };

int main(int argc, char** argv)
{
	int test_case;
	int T;
	cin >> T;

	for (test_case = 1; test_case <= T; ++test_case)
	{
		char tr;
		int tr1;
		cin >> tr >> tr1;
		int num_all;
		cin >> num_all;
		char temp[3];
		for (int i = 0; i < num_all; i++) {
			cin >> temp[0] >> temp[1] >> temp[2];
			for (int j = 0; j < 10; j++) {
				if (number[j][0] == temp[0] && number[j][1] == temp[1] && number[j][2] == temp[2]) {
					num_cnt[j]++;
					break;
				}
			}
		}

		cout << "#" << test_case << "\n";
		for (int i = 0; i < 10; i++) {
			while (num_cnt[i]) {
				cout << number[i] << " ";
				num_cnt[i]--;
			}
		}
	}
	return 0;//정상종료시 반드시 0을 리턴해야합니다.
}